nipype/pipeline/engine/utils.py

```diff
@@ -1314,18 +1314,17 @@ def export_graph(graph_in,
 def format_dot(dotfilename, format='png'):
     """Dump a directed graph (Linux only; install via `brew` on OSX)"""
     if format != 'dot':
-        cmd = 'dot -T%s -O \'%s\'' % (format, dotfilename)
+        dot_base =  os.path.splitext(dotfilename)[0]
+        formatted_dot = '{}.{}'.format(dot_base, format)
+        cmd = 'dot -T{} -o"{}" "{}"'.format(format, formatted_dot, dotfilename)
```

Comment:
Returning `formatted_dot` breaks if `format == 'dot'`. One option is to revert the return line, and update these three lines to:

```Python
dot_base =  os.path.splitext(dotfilename)[0]
orig_dot = dotfilename
dotfilename = '{}.{}'.format(dot_base, format)
cmd = 'dot -T{} -o"{}" "{}"'.format(format, dotfilename, orig_dot)
```

Reply:
thanks for the advice, but I would recommend the current approach. Having `dotfilename` mean both the original and the formatted filename is confusing, but I would rather not construct it implicitly inline with some other operation for the sake of clarity.

If the filename for some reason can't be split, I'd much rather like the traceback to serve me up this line directly:

```
dot_base =  os.path.splitext(dotfilename)[0]
```

Reply:
The issue is that you're returning a value from an unassigned name in the case where `format == 'dot'`. If you don't want to reassign a variable, you can return the `formatted_dot` from within the `if` block, or `return dotfilename if format == 'dot' else formatted_dot`, or whatever, but the tests will need to pass before we can merge this.